### PR TITLE
Backend on docker

### PR DESCRIPTION
**Pull request by yvonnegitau**
# Backend on Docker

The goal of this task is to containerise the backend. This will simplify working with the backend and in the end simplify setting up Dalgo. Below are the changes made:

- Created two Dockerfiles, one for building the main image and another as an example on how to customize the main image for different deployments.
- Created a Docker Compose file. One for dev (`docker-compose.dev.yml`) environment where the developer uses postgres image as a database and the main (`docker-compose.yml`) using external database.
-  Added a mount folder. This is where all files that will be specific to a developer will be stored. An example is the `whitelist.py` file.
- Added documentation on how to run the backend using docker in Readme.
- We have added Redis change which will enable developer to add REDISHOST and REDISPORT

## Potential Breaking changes
Changes to the `gunicorn-log.conf` may affect applications already using it. Changed the path from absolute to relative considering developers may all not have the code in the same path.

### CODE REVIEW

File: ddpui/utils/taskprogress.py
Comment (fatchat) [inline ddpui/utils/taskprogress.py:19]: this is from an older version, could you update to the new logic please? the PR diff has a comment explaining why

File: ddpui/settings.py
Comment (fatchat) [inline ddpui/settings.py:47]: what are these two hosts?

File: ddpui/settings.py
Comment (fatchat) [inline ddpui/settings.py:65]: what is this host?

**Comment by fatchat** [comment]
initial read-through looking good, looking forward to running it later this weekend

**Comment by fatchat** [comment]
also `dbt_api.post_dbt_makedocs` still refers to `Redis`, this should be updated 

(actually we don't use this endpoint anymore but i still want to keep it around until i know we'll never use it)

**Comment by fatchat** [comment]
@yvonnegitau We needed your redis client for another large PR so we pulled it out into its own PR and merged it. Would you please pull into this docker PR thank you

**Comment by yvonnegitau** [comment]
> @yvonnegitau We needed your redis client for another large PR so we pulled it out into its own PR and merged it. Would you please pull into this docker PR thank you

ok

File: ddpui/settings.py
Comment (yvonnegitau) [reply ddpui/settings.py:47]: This is our hosts which I will remove. I think in the next PRs we will move this to .env since it will be different for each deployment

File: ddpui/settings.py
Comment (yvonnegitau) [reply ddpui/settings.py:65]: This is ours but removing it

**Comment by fatchat** [comment]
notes to self:
- existing code changes are only re-ordering imports so merging is safe
- asked rishabh to try running it will wait for his response

**Comment by fatchat** [comment]
@yvonnegitau we also need to start a celery worker and celery beat where should that be done

**Comment by fatchat** [comment]
one more thing @yvonnegitau we are moving from wsgi to asgi, i think that will affect your setup you may want to wait until #617 has been merged

should be this weekend or early next week

**Comment by yvonnegitau** [comment]
> one more thing @yvonnegitau we are moving from wsgi to asgi, i think that will affect your setup you may want to wait until #617 has been merged
> 
> should be this weekend or early next week
Ok. I will be on the lookout

**Comment by yvonnegitau** [comment]
> @yvonnegitau we also need to start a celery worker and celery beat where should that be done

I will add the services. Need to test them. Could you confirm the startup command are `celery -A celery_app.celery worker` and `celery -A celery_app.celery beat`

**Comment by fatchat** [comment]
We run
`celery -A ddpui beat`
`celery -A ddpui worker -n ddpui`

**Comment by yvonnegitau** [comment]
> We run `celery -A ddpui beat` `celery -A ddpui worker -n ddpui`

ok. Let me test it

**Comment by yvonnegitau** [comment]
> > We run `celery -A ddpui beat` `celery -A ddpui worker -n ddpui`
> 
> ok. Let me test it

@fatchat This is completed

**Review by fatchat — APPROVED** [review]

**Pull request merged**
